uvicorn/supervisors/multiprocess.py

```diff
@@ -90,9 +90,13 @@ def kill(self) -> None:
         # In Unix, the method will send SIGKILL to the process.
         self.process.kill()
 
-    def join(self) -> None:
+    def join(self, timeout: float | None = None) -> None:
         logger.info(f"Waiting for child process [{self.process.pid}]")
-        self.process.join()
+        self.process.join(timeout)
+        # Timeout, kill the process
+        while self.process.exitcode is None:
+            self.process.kill()
```

Comment:
The reason why CI failed is that this is not covered by the test. But I don't know how to design a process that will be 100% stuck.

Reply:
It's okay to add the pragma here.